kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/logstore/LogStoreProvider.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.logstore;
+
+import java.util.*;
+
+import io.delta.storage.*;
+import org.apache.hadoop.conf.Configuration;
+
+import io.delta.kernel.defaults.internal.DefaultTableClientErrors;
+
+/**
+ * Utility class to provide the correct {@link LogStore} based on the scheme of the path.
+ */
+public class LogStoreProvider {
+
+    // Supported schemes per storage system.
+    private static final Set<String> s3Schemes = unmodifiableSet("s3", "s3a", "s3n");
+    private static final Set<String> azureSchemes =
+            unmodifiableSet("abfs", "abfss", "adl", "wasb", "wasbs");
+    private static final Set<String> gsSchemes = unmodifiableSet("gs");
+
+    /**
+     * Get the {@link LogStore} instance for the given schema and configuration. Callers can set
+     * {code io.delta.kernel.logStore.class} to specify the LogStore implementation to use. If not
```

Comment:
why support both `io.delta.kernel.logStore.class` and `io.delta.kernel.logStore.<scheme>.impl`?

Reply:
This should be `io.delta.kernel.logStore.<scheme>.impl`. Updated the javadoc